### PR TITLE
mongodb fix python acceptance tests

### PR DESCRIPTION
**Pull request by jpefaur**
## What
get mongo db python acceptance tests to pass

## How
- disabled all tests related to full refresh syncs since connector doesn't support them
- enabled spec test, disabling backwards compatibility tests for previous version of the connector since it didn't support `spec`
- disabled discover test since it was failing on setup because previous version of the connector returns null when discover is called
- won't enable incremental tests for now since connector doesn't fully support it yet

## More about incremental syncs

We found a few issues while running incremental sync tests:
1. **MongoServerUnavailableException**: after reading some records from the database, eventually we get an error that looks like this:
```
com.mongodb.MongoServerUnavailableException: The server at cluster0-shard-00-02.iqgf8.mongodb.net:27017 is no longer available
```
when calling `hasNext` on the `MongoCursor` https://github.com/airbytehq/airbyte/pull/29066/files#diff-d08766f2cf8db220a265614851f7810de5acc2463140e4cbf47d0f73c1797e3dR52. I tested that we are able to read some records before this happens, but can't really tell why we get this error. It also seems to happen while iterating over documents in the first batch. It would make more sense if this happened in the first batch after trying to fetch the first document, but it happens in the first batch after successfully fetching a few documents.

We got around this by doing [this](https://github.com/airbytehq/airbyte/pull/29066/commits/75a684bfdd94c9f0b621a0352e94ab1fdede3924). It's still unclear why we get the exception

2. **ClassCastException**: Running the tests after fixing the above we get the following:
```
java.lang.ClassCastException: class java.lang.String cannot be cast to class org.bson.types.ObjectId
```
this is because the collection we are using to test has documents with `_id` that can be either `String` or `ObjectId`, which is something we are not planning to support. To get around it locally, I added a `.filter(Filters.type("_id", BsonType.OBJECT_ID))` filter to the mongo query. Not going to do anything about this for now, but we are probably going to need a new collection for this test

3. **AssertionError**: Running the tests after "fixing" the above, tests fail with the following message:
```
AssertionError: Second incremental sync should produce records older or equal to cursor value from the state. Stream: listingsAndReviews
E               assert False
E                +  where False = compare_cursor_with_threshold('614a0160ba0f4b8a31f6278d', '614a4fc7e8f281851051abfa', 0)
```
this is actually expected since the connector doesn't really support incremental syncs ->
- first sync runs and saves state
- second sync runs ignoring the saved state

since this is expected, I'm going to disable the incremental tests for now

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`1cf4d9c60a`](https://github.com/airbytehq/airbyte/pull/29243/commits/1cf4d9c60a0e297cad4c53423ad2c3639826317c)) - ❌

⏲️  Total pipeline duration: 06mn27s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ❌        |
| Connector version semver check                                                         | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jose_mongodb-fix-python-acceptance-tests/1691605621/1cf4d9c60a0e297cad4c53423ad2c3639826317c/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1cf4d9c60a0e297cad4c53423ad2c3639826317c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py [hunk 1]
```diff
@@ -323,6 +323,13 @@ def previous_discovered_catalog_fixture(
             )
             return None
         catalogs = [message.catalog for message in output if message.type == Type.CATALOG]
+
+        if catalogs[-1] is None:
```
[the remark this answers is not in the document]
Comment (colesnodgrass) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py:327]: I vote to disable it with a nice, big `TODO` comment (or issue), so we don't forget

**Review by colesnodgrass — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`86ff2bd367`](https://github.com/airbytehq/airbyte/pull/29243/commits/86ff2bd367ec3ab965f2c27922b140b973c3ff47)) - ❌

⏲️  Total pipeline duration: 07mn42s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ❌        |
| Connector version semver check                                                         | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jose_mongodb-fix-python-acceptance-tests/1691619899/86ff2bd367ec3ab965f2c27922b140b973c3ff47/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:86ff2bd367ec3ab965f2c27922b140b973c3ff47)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Comment by octavia-squidington-iii** [comment]
##  source-mongodb-internal-poc test report (commit [`481e04b9a5`](https://github.com/airbytehq/airbyte/pull/29243/commits/481e04b9a5c29fe74d4ebdfcf3ed74db49fed340)) - ❌

⏲️  Total pipeline duration: 07mn42s 

| Step                                                                                   | Result   |
|:---------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-mongodb-internal-poc/metadata.yaml     | ❌        |
| Connector version semver check                                                         | ✅        |
| QA checks                                                                              | ❌        |
| Build connector tar                                                                    | ✅        |
| Build source-mongodb-internal-poc docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:source-mongodb-internal-poc:integrationTest | ✅        |
| Acceptance tests                                                                       | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/jose_mongodb-fix-python-acceptance-tests/1691642599/481e04b9a5c29fe74d4ebdfcf3ed74db49fed340/source-mongodb-internal-poc/0.0.1/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:481e04b9a5c29fe74d4ebdfcf3ed74db49fed340)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-mongodb-internal-poc test
```

**Pull request merged**